### PR TITLE
Laboratory Work #1, Ksenia Eremeeva - 20FPL3

### PR DESCRIPTION
**Pull request by KseniaEremeeva**

### CODE REVIEW

File: scrapper.py [hunk 1]
```diff
@@ -35,13 +39,45 @@ def find_articles(self):
...
-        pass
 
-    def get_search_urls(self):
+        url = 'https://aif.ru/dosug'
+        response = requests.get(url, headers={
+            "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36",
+            "accept-encoding": "gzip, deflate, br", "accept-language": 'en-US,en;q=0.9,ru-RU;q=0.8,ru;q=0.7,fr;q=0.6',
+            "cookie": "__ddg1=3z7ohwF248ukR81grhvj; CookieMessenger=; _ga=GA1.2.1866016929.1646422780; _ym_d=1646422780; _ym_uid=1620758162826342715; __ddgid=MhwOCqWUpVMS0HAn; subscription_popup_min_state=week_view1; subscription_popup_state=%7B%22key%22%3A%22week_view1%22%2C%22timeout%22%3A1647599958264%7D; __ddgmark=JrhVBqNQJzM3xeRX; __ddg5=T7GglOZhSXdoo9HS; aif_sid=b806513bc5db4b27668b98a5d3e86f74; _gid=GA1.2.1353469175.1647194832; _ym_isad=1; _ym_visorc=b; _gat_gtag_UA_3672159_1=1"},
+                                allow_redirects=True)
+        with open('index.html', 'w', encoding='utf-8') as file:
```
Comment (dmitry-uraev) [inline scrapper.py:49]: I think you may avoid downloading and writing HTML files. All you need is just URL

File: scrapper.py [hunk 2]
```diff
@@ -35,13 +39,45 @@ def find_articles(self):
...
+            try:
+                # print(url + link['href'])
+                seed_urls.append(link['href'])
+            except (KeyError, urllib.error.URLError, urllib.error.HTTPError):
+                print('Found incorrect link')
+
+            for elements in seed_urls:
+                if ('http://' and 'https://') not in elements:
+                    seed_urls.remove(elements)
+        return seed_urls
```
Comment (dmitry-uraev) [inline scrapper.py:80]: do you have a link limit or you just take all links?

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Good start. Keep going

File: scrapper.py (hunk 2 above)
Comment (KseniaEremeeva) [reply scrapper.py:80]: Now it's just all valid links

File: scrapper.py [hunk 3]
```diff
@@ -25,39 +30,126 @@ class Crawler:
...
+                break
+
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+
+        response = requests.get(self.seed_urls, headers={
```
Comment (dmitry-uraev) [inline scrapper.py:55]: can you propose another efficient way of storing headers?

File: scrapper.py [hunk 4]
```diff
@@ -25,39 +30,126 @@ class Crawler:
...
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        text_of_article_bs = article_bs.find('span')
+        self.article.text = text_of_article_bs.text
+
+    def parse(self):
+        response = requests.get(self.article_url, headers={
```
Comment (dmitry-uraev) [inline scrapper.py:96]: same as question above

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Make sure you pass tests before Friday assignment.

**Pull request closed without merging**